xbmc/guilib/FFmpegImage.cpp

```diff
@@ -41,10 +42,33 @@ struct MemBuffer
   size_t pos;
 };
 
+struct ThumbDataManagement
+{
+  uint8_t* intermediateBuffer = nullptr; // gets av_alloced
+  AVFrame* frame_input = nullptr;
+  AVFrame* frame_temporary = nullptr;
+  SwsContext* sws = nullptr;
+  AVCodecContext* avOutctx = nullptr;
+  AVCodec* codec = nullptr;
+  ~ThumbDataManagement()
+  {
+    av_free(intermediateBuffer);
+    intermediateBuffer = nullptr;
+    av_frame_free(&frame_input);
+    frame_input = nullptr;
+    av_frame_free(&frame_temporary);
+    frame_temporary = nullptr;
+    avcodec_close(avOutctx);
+    avcodec_free_context(&avOutctx);
+    avOutctx = nullptr;
+    sws_freeContext(sws);
+    sws = nullptr;
+  }
+};
 static size_t Clamp(int64_t newPosition, size_t bufferSize, size_t offset)
 {
   size_t clampedPosition = 0;
-  if (newPosition < 0 || offset + newPosition < 0)
+  if (newPosition < 0)
```

Comment:
https://github.com/fritsch/xbmc/commit/3900a8ead92cea923e29918912d5e732a6674c86 <- did you think that way?